app/controllers/support_interface/references_controller.rb

```diff
@@ -12,10 +12,14 @@ def confirm_cancel
       redirect_to support_interface_application_form_path(@reference.application_form)
     end
 
-    def reinstate; end
+    def reinstate
+      render_404 and return unless ReferenceActionsPolicy.new(@reference).reinstatable?
+    end
 
     def confirm_reinstate
-      RequestReference.new.call(@reference)
+      render_404 and return unless ReferenceActionsPolicy.new(@reference).reinstatable?
```

Comment:
It's failing for me 🤔 

<kbd>![Kapture 2021-02-25 at 17 21 06](https://user-images.githubusercontent.com/38078064/109191571-36009000-778e-11eb-8838-89ecb47a00ee.gif)</kbd>

```
  ↳ app/services/reinstate_reference.rb:5:in `call'
  ApplicationChoice Update All (2.3ms)  UPDATE "application_choices" SET "updated_at" = $1 WHERE "application_choices"."application_form_id" = $2  [["updated_at", "2021-02-25 16:19:01.526446"], ["application_form_id", 860]]
  ↳ app/models/concerns/published_in_api.rb:6:in `block (2 levels) in <module:PublishedInAPI>'
   (0.4ms)  SELECT COUNT(*) FROM "references" WHERE "references"."application_form_id" = $1 AND "references"."feedback_status" = $2  [["application_form_id", 860], ["feedback_status", "feedback_provided"]]
  ↳ app/models/application_form.rb:359:in `enough_references_have_been_provided?'
Completed 500 Internal Server Error in 33ms (ActiveRecord: 9.9ms | Allocations: 15418)


ApplicationReference#648 can't be requested excluded from capture: DSN not set
  
RuntimeError (ApplicationReference#648 can't be requested):
  
app/services/request_reference.rb:6:in `call'
app/services/reinstate_reference.rb:12:in `call'
app/controllers/support_interface/references_controller.rb:22:in `confirm_reinstate'
app/middlewares/vendor_api_request_middleware.rb:27:in `_call'
app/middlewares/vendor_api_request_middleware.rb:22:in `call'
app/middlewares/redirect_to_service_gov_uk_middleware.rb:11:in `call'
Started GET "/support/applications/860" for ::1 at 2021-02-25 17:19:29 +0000
```